src/api/db/build.ts

```diff
@@ -1,17 +1,23 @@
 import { Build, BuildRun, Job } from './model';
 import { getLastRun } from './job';
 
-export function getBuilds(limit: number, offset: number): Promise<any> {
+export function getBuilds(userId: number, limit: number, offset: number): Promise<any> {
   return new Promise((resolve, reject) => {
     new Build()
       .query(q => q.orderBy('id', 'DESC').offset(offset).limit(limit))
-      .fetchAll({ withRelated: ['repository', 'jobs.runs'] })
+      .fetchAll({ withRelated: ['repository.permissions', 'jobs.runs'] })
       .then(builds => {
         if (!builds) {
           reject();
         }
 
         builds = builds.toJSON();
+        builds = builds.filter(b => {
```

Comment:
this is not okay for many reasons:
- limit and offset will not work properly
- you are fetching all results and then filtering array by permissions, so its a high possibility that you fetch data from database that you will not need. this is overkill.

what I would suggest here is simply like you did here https://github.com/Izak88/abstruse/blob/481faefd50f97667618e0660e94a5ef0a249f7dc/src/api/db/repository.ts#L100. that would be the right way.